### ISSUE TITLE
use Pipeline instead of `HttpClient` for `azure_identity`

### ISSUE THREAD

**Issue by cataggar** [Azure/azure-sdk-for-rust#1015]
> @bmc-msft I thought we were moving away from new_http_client and moving to Pipeline?

> @rylev I guess the question is whether identity should also go through a Pipeline. It makes sense in a way. The challenge is that there is specific logic the oauth client handles for us that a simple transport policy would not do.

https://github.com/Azure/azure-sdk-for-rust/pull/997#pullrequestreview-1074407148

**Comment by ctaggart** [comment]
@minghuaw, I'm not sure. Right now, the Pipeline is pretty specific to HTTP. cc @rylev

**Comment by rylev** [comment]
Indeed, the current pipeline architecture assumes HTTP. I imagine we'll want pipeline-like architecture for AMQP based clients as well, but what that should look like is not something I've personally given much thought to.
